src/Runtime/CMakeLists.txt

```diff
@@ -114,3 +120,30 @@ llvm_update_compile_flags(PyRuntime)
 install(TARGETS PyRuntime
   DESTINATION lib
   )
+
+pybind11_add_module(PyCompileAndRuntime PyOMCompileExecutionSession.cpp)
+add_dependencies(PyCompileAndRuntime onnx_proto)
+target_compile_options(PyCompileAndRuntime
+  PRIVATE
+  $<$<OR:$<CXX_COMPILER_ID:Clang>,$<CXX_COMPILER_ID:AppleClang>,$<CXX_COMPILER_ID:GNU>>:-frtti -fexceptions>
+  $<$<CXX_COMPILER_ID:MSVC>:/EHsc /GR>
+  )
+target_compile_definitions(PyCompileAndRuntime
+  PRIVATE
+  $<TARGET_PROPERTY:onnx,COMPILE_DEFINITIONS>
+  )
+target_include_directories(PyCompileAndRuntime
+  PRIVATE
+  ${ONNX_MLIR_SRC_ROOT}/include
+  $<TARGET_PROPERTY:onnx,INCLUDE_DIRECTORIES>
+  )
+target_link_libraries(PyCompileAndRuntime
```

Comment:
Why aren't you linking this against PyCompile  and PyRuntime?

Reply:
Because `PyCompile` and `PyRuntime` are not normal C++ libraries. Furthermore, they are not even normal `shared libraries` which means this [solution](https://github.com/pybind/cmake_example/issues/11) I have tried before does not work. I have also tried another [solution](https://github.com/pybind/pybind11/issues/2056) to combine two pybind11-wrapped projects. It also does not work either as it is a little bit different from our situation. We need to have two objects of PyCompile and PyRuntime inside the new class and then use the functions inside the objects instead of just calling functions from another pybind11-generated Python binding. Also, I did not get enough information from the [llvm-project pybind11_add_module](https://github.com/llvm/llvm-project/blob/24989646a5216cc6f8c6d336a17e7cb2f5c60454/mlir/cmake/modules/AddMLIRPython.cmake#L638). Maybe I am looking into some wrong places.